### ISSUE TITLE
Compiler Can't Find LINQ Extensions After Calling AsAsyncEnumerable

### ISSUE THREAD

**Issue by Cooksauce** [dotnet/efcore#18588]
When calling the `AsAsyncEnumerable()` on an `IQueryable<T>`, the compiler can't find any of the LINQ extension methods on the returned `IAsyncEnumerable<T>`.

However,
When calling `ToAsyncEnumerable()` the LINQ methods are found. 

Everything I see shows `AsAsyncEnumerable()` to be the correct method to call.

Is this the case? Or should I be calling `ToAsyncEnumerable()`?

Usage:
```
IQueryable<MyType> myTypes = _dbContext.MyTypes;
var projected = myTypes.AsAsyncEnumerable().Select(x => x);
```

Intellisense message:
Error CS1061
`'IAsyncEnumerable<MyType>' does not contain a definition for 'Select' and no accessible extension method 'Select' accepting a first argument of type 'IAsyncEnumerable<MyType>' could be found (are you missing a using directive or an assembly reference?)`

### Further technical details

EF Core version: 3.0
Target framework: NET Core 3.0

**Comment by ajcvickers** [comment]
@Cooksauce This is something that could be implemented in the [base class libraries](https://github.com/dotnet/corefx) but isn't. The external [System.Interactive.Async package](nuget.org/packages/System.Interactive.Async/) can be used, but be aware that it [doesn't play nice with its namespaces](https://github.com/aspnet/EntityFrameworkCore/issues/18124).

**Comment by Cooksauce** [comment]
@ajcvickers Just an FYI - We determined this was caused by a NuGet Package, [Google.Cloud.Storage.V1 - 2.3.0](https://www.nuget.org/packages/Google.Cloud.Storage.V1/2.3.0), which depends on [System.Interactive.Async - 3.1.1](https://www.nuget.org/packages/System.Interactive.Async/3.1.1). Depending on in [System.Interactive.Async - 4.0.0](https://www.nuget.org/packages/System.Interactive.Async/4.0.0) directly from the project fixes the issue, and results in the namespace issues you mentioned.

I did just want to confirm that `AsAsyncEnumerable()` is the EF Core recommended method to use to evaluate in memory. (vs `ToAsyncEnumerable()`) Is that correct?
